smartsim/_core/mli/infrastructure/storage/featurestore.py

```diff
@@ -26,22 +26,32 @@
 
 import typing as t
 from abc import ABC, abstractmethod
-
-from pydantic import BaseModel, Field
+from dataclasses import dataclass
 
 from smartsim.log import get_logger
 
 logger = get_logger(__name__)
 
 
-class FeatureStoreKey(BaseModel):
+@dataclass(frozen=True)
```

Comment:
Do we want ``frozen=True``?  I'm not familiar with the ``BaseModel`` -- is that frozen by default?

If frozen is not used, then I would assume that we need to add setter methods to check that any updates adhere to non-empty string

Reply:
`BaseModels` are not frozen by default, but I think we definitely want a frozen dataclass because we don't want `FeatureStoreKey` instances to be updated.